core/src/main/java/org/apache/spark/unsafe/map/BytesToBytesMap.java

```diff
@@ -816,6 +816,10 @@ public boolean append(Object kbase, long koff, int klen, Object vbase, long voff
           } catch (SparkOutOfMemoryError oom) {
             canGrowArray = false;
           }
+        } else if (numKeys >= growthThreshold && longArray.size() / 2 >= MAX_CAPACITY) {
+          // The map needs to grow, but this would cause it to exceed MAX_CAPACITY. Prevent the map
+          // from accepting any more new elements.
+          canGrowArray = false;
```

Comment:
The map won't exceed `MAX_CAPACITY`, I think. This just prevents the map accepting new elements more early.

Reply:
Hmm, perhaps the comment's wording is confusing. I intended to convey that, if the map were to grow, then its capacity would exceed MAX_CAPACITY. Therefore it cannot grow and we must prevent it from accepting any more new elements.

I reworded the comment to clarify this.